parser/scanner.l

```diff
@@ -133,21 +133,26 @@ func (l *lexer) next() int {
 	return l.c
 }
 
-func (l *lexer) err0(ln, c int, args ...interface{}) {
+func (l *lexer) err0(ln, c int, arg interface{}) {
 	var argStr string
-	if len(args) > 0 {
-		argStr = fmt.Sprintf(" " + args[0].(string), args[1:]...)
+	switch x := arg.(type) {
+		case error:
+			argStr = " " + x.Error()
+		case string:
+			argStr = " " + x
 	}
+
 	err := errors.New(fmt.Sprintf("line %d column %d near \"%s\"", ln, c, l.val) + argStr)
 	l.errs = append(l.errs, err)
 }
 
-func (l *lexer) err(s string, args ...interface{}) {
-	l.err0(l.line, l.col, args...)
+func (l *lexer) err(arg interface{}) {
```

Comment:
we can support `errf(format string, args ...string)`.

btw, I think `err(args ...interface{})` may be the correct format, you can see

https://golang.org/pkg/fmt/#Sprint